src/main/java/org/mockito/internal/exceptions/stacktrace/DefaultStackTraceCleaner.java

```diff
@@ -24,13 +24,20 @@ public boolean isIn(StackFrameMetadata e) {
     private boolean isIn(String className) {
         if (isFromMockitoRunner(className) || isFromMockitoRule(className)) {
             return true;
-        } else if (isMockDispatcher(className) || isFromMockito(className)) {
+        } else if (isMockDispatcher(className)
+                || isFromMockito(className)
+                || isMethodHandle(className)) {
```

Comment:
Fixing `PartialMockingWithSpiesTest`, @TimvdLippe apologies, but another tricky one: the inline mock maker uses `MethodHandle` to dispatch method calls and this fact leaks into the stacktrace. I have added the filtering here, but we may unintentionally filter out the user's stacktrace as well (fe if test or code under test uses `MethodHandle`), what do you think?

Reply:
Wouldn't it be an option to filter all frames after a given frame here? That way, "our" handle frames should be discovered.

Reply:
> Wouldn't it be an option to filter all frames after a given frame here? That way, "our" handle frames should be discovered.

@raphw I looked at current `StackTraceCleaner`, it operates one frame at a time: there is no context with respect what previous (or next) stack frames are. I think in general it is possible to do, but needs some redesign.